front/components/spaces/SpaceDataSourceViewContentList.tsx

```diff
@@ -211,8 +211,7 @@ export const SpaceDataSourceViewContentList = ({
     [setPagination, setViewType, viewType, pagination.pageSize]
   );
 
-  const isServerPagination =
-    !isManaged(dataSourceView.dataSource) && !dataSourceSearch;
+  const isServerPagination = !isManaged(dataSourceView.dataSource);
```

Comment:
I'm not sure to fully understand the previous behaviour, but do you think we need to call `handleViewTypeChange` when setting the `pagination`?

Reply:
Damn, sorry was trying to fix something else and committed in this PR, reverting

Reply:
There's indeed a problem with pagination, but I'll address this in another PR 👌🏼

Reply:
you can explain to me then 😉